#72 migrate forgot pass to angular

Sheptytskyid opened the pull request.

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.travis.yml
#	pom.xml
#	sonar-project.properties
#	src/main/java/net/greatstart/model/Contact.java

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added PasswordReset controller and service on front - end

Commit:
BE controller returns status and FE parses it

Commit:
modal forgotPassword works

Commit:
fixed JavaMail issues

Commit:
token validation works

Commit:
finished

Sheptytskyid commented:
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

Sheptytskyid commented on src/main/java/net/greatstart/controllers/PasswordResetController.java:
```diff
@@ -1,117 +1,90 @@
-            model.addObject(MESSAGE, messages.getMessage("token.error", null, locale) + result);
-            model.setViewName("login/login");
+            model.setViewName("redirect:/");
+            //TODO: implement displaying errors, maybe in some generic error page
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

Commit:
removed user authority "change password"
token is now validated when submitting new password

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into #72_migrate_forgot_pass_to_angular

# Conflicts:
#	src/main/resources/application.properties

Commit:
updated PasswordResetControllerTests(all tests) and SecurityServiceTest(getUserByToken)

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into #72_migrate_forgot_pass_to_angular

# Conflicts:
#	src/main/java/net/greatstart/controllers/UserController.java

Commit:
replaced meta mail with Ukr net

Commit:
replaced Ukr.net with Gmail

Commit:
fixed mail address

holkwen commented:
Я думаю, модальные окна можно закрывать. Например, при показе окна восстановления пароля окно логина становится бессмысленным, его можно закрыть. После отсылки линка можно выводить кнопку "ОК" для закрытия окна восстановления.

А так вроде всё нормально.

Commit:
login modal closes when forgot pass modal opens

Commit:
added "close" button to ForgotPassword.html that appears after successful email sending

Sheptytskyid commented:
@holkwen Done!

zozich commented on src/main/java/net/greatstart/controllers/PasswordResetController.java:
```diff
@@ -1,117 +1,79 @@
-            return model;
-        }
-        User user = userService.getUserByEmail(dtoUser.getEmail());
+    public ResponseEntity<String[]> processEmail(HttpServletRequest request, @RequestParam("email") String email) {
```
1. No need to indicate param name if it has the same name as imported variable
2. Why do you return String[]?

zozich commented on src/main/java/net/greatstart/controllers/PasswordResetController.java:
```diff
@@ -1,117 +1,79 @@
-            model.addObject(MESSAGE, messages.getMessage("message.resetPassword.success", null, locale));
-        }
-        return model;
+    @ExceptionHandler(ServiceException.class)
```
You should have one generic exception handler. There is a task for that

zozich commented on src/test.java.net.greatstart.model/net/greatstart/model/ContactTest.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Test
+    public void addressShouldEqual() {
+        assertEquals("StreetTest", contact.getAddress());
```
WTF?

zozich commented on src/test/java/net/greatstart/controllers/PasswordResetControllerTest.java:
```diff
@@ -61,134 +46,61 @@
+        mvc.perform(get("/user/resetPassword").param("email", VALID_EMAIL)).andExpect(status().isBadRequest());
     }
 
     @Test(timeout = 2000)
```
Why do you have timeouts here?

zozich commented on src/main/java/net/greatstart/controllers/PasswordResetController.java:
```diff
@@ -1,117 +1,79 @@
         }
         String passwordResetToken = securityService.createPasswordResetToken(user).getToken();
-        String url = request.getHeader("origin");
+        String url = request.getHeader("referer");
```
There should be no business logic in controller

zozich reviewed (approved):
This should be reworked. Approving only because of lack of time

Commit:
removed business logic from PasswordResetController on Java
status messages are now generated by the front-end controller
updates unit tests

Sheptytskyid commented:
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/resources/static/views/main/PasswordResetController.js:
```diff
@@ -0,0 +1,55 @@
+            $scope.submitted = true;
+            $scope.message = null;
+            $scope.error = null;
+            var data = passwordResetService.get({email:$scope.email}, function(){
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "data" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'data' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AUnusedVariable)

Commit:
fixed sonar issue in PasswordResetCOntroller.js
